paddle/fluid/platform/device_context.cc

```diff
@@ -407,17 +407,32 @@ thread_local size_t cur_mkldnn_session_id = kMKLDNNSessionID_Default;
 // - For fixed-shape, it's a null string in default.
 // - For dynamic-shape, it's user specific.
 thread_local std::string cur_input_shape_str = "";
+// the cache size of different input shapes for MKLDNN.
+// Default 1 means fixed input shape, not dynamic shape.
+thread_local int cur_input_shape_cache_size = 1;
 }  // namespace
 
 void set_cur_mkldnn_session_id(size_t sid) { cur_mkldnn_session_id = sid; }
 size_t get_cur_mkldnn_session_id(void) { return cur_mkldnn_session_id; }
 void set_cur_input_shape_str(std::string input_shape_str) {
   cur_input_shape_str = input_shape_str;
 }
-std::string get_cur_input_shape_str(void) { return cur_input_shape_str; }
+void set_cur_input_shape_cache_size(int input_shape_cache_size) {
+  cur_input_shape_cache_size = input_shape_cache_size;
+}
 
 void MKLDNNDeviceContext::ResetBlobMap() const { p_blobmap_->clear(); }
 
+size_t MKLDNNDeviceContext::GetShapeBlobSize(int mkldnn_session_id) const {
```

Comment:
why need a explicit input parameter mkldnn_session_id? can we just get it from thread_local? seems there is no case for user getting shapeblobsize for other session id with different id in thread_local

Reply:
> can we just get it from thread_local

Could you give more details on how to get it from thread_local?

> why need a explicit input parameter mkldnn_session_id? 

if we don't have `mkldnn_session_id` parameter, line428 `pMap->find(mkldnn_session_id)` could not find. Or do you mean there is only one session_id in `pMap`?

Reply:
mkldnn_session_id is already stored in cur_mkldnn_session_id, it is a thread_local variable, so if user call it in same thread with predictor.run, it can easily get id from cur_mkldnn_session_id.

Reply:
if user call getblobsize in same thread with predictor.run, then code is like:
```
  BlobMap* pMap = p_blobmap_.get();
  auto map_it = pMap->find(cur_mkldnn_session_id);
  if (map_it == pMap->end()) {
    LOG(FATAL) << "MKLDNNDeviceContext don't find mkldnn_session_id : "
               << mkldnn_session_id;
  }
  return map_it->second->size();
}
```
If we still want user to call this function in other threads, then is parameter is necessary.

Reply:
Got it. I will change it.